question_framework/validation/validation.py

```diff
@@ -57,3 +57,17 @@ def ip_range(x: str) -> bool:
         return False
     ip1, ip2 = split_ips
     return is_ip_check(ip1) and is_ip_check(ip2)
+
+def date_validation(x):
+  date_format = '%Y-%m-%d'
```

Comment:
Hardcoding the date format limits the number of scenarios you can use this validation.

I think you should make this a function factory that takes a date format and creates/returns a function that utilizes that format to determine if the input is correctly formatted. The return function should take in a string and output a bool.

This function from the function factory should ONLY return  True or False. 

There are some examples of a function factory in the validation.py file.